### PR TITLE
fix: confusing nil in direct interface with typed nil

### LINKED ISSUE #376: Bug: struct with map (the only one field) encoded to `null`
```golang
package main

import (
	"fmt"
	"testing"

	"github.com/goccy/go-json"
)

func main() {
	type Container struct {
		V any `json:"value"`
	}
	type MapOnly struct {
		Map map[string]int64 `json:"map"`
	}

	b, err := json.Marshal(Container{MapOnly{}})
	if err != nil {
		panic(err)
	}

	fmt.Println(string(b))
}
```

output: `{"value":null}`

### PR DESCRIPTION
**Pull request by orisano**
fix #376

### CODE REVIEW

File: internal/cmd/generator/vm.go.tmpl [hunk 1]
```diff
@@ -193,7 +194,8 @@ func Run(ctx *encoder.RuntimeContext, b []byte, codeSet *encoder.OpcodeSet) ([]b
 				ifacePtr = iface.ptr
 				typ = iface.typ
 			}
-			if ifacePtr == nil {
+			isDirectedNil := typ != nil && typ.Kind() == reflect.Struct && !runtime.IfaceIndir(typ)
```
Comment (Trim21) [inline internal/cmd/generator/vm.go.tmpl:197]: maybe check this condition in `if ifacePtr == nil {` ?

It will be executed every times then for is not nil

Comment (orisano) [reply internal/cmd/generator/vm.go.tmpl]: nice catch!

**Comment by orisano** [comment]
ping @goccy

**Review by goccy — APPROVED** [review]
LGTM !

**Pull request merged**